manager/integration/tests/test_scheduling.py

```diff
@@ -2504,3 +2504,50 @@ def best_effort_data_locality_test(client, volume_name):  # NOQA
         time.sleep(RETRY_INTERVAL)
 
     assert volume.replicas[0]['hostId'] == self_node
+
+@pytest.mark.csi  # NOQA
+def test_storage_capacity_aware_pod_scheduling(client, core_api, storage_class, statefulset):  # NOQA
+    """
+    Test that kube-scheduler is aware of storage capacity available on each
+    node when scheduling pods using a StorageClass with volumeBindingMode set
+    to 'WaitForFirstConsumer'.
+
+    1. Reduce the schedulable storage on all nodes (except the current node)
+       to 10Gi.
+    2. Create a new StorageClass with volumeBindingMode set
+       to 'WaitForFirstConsumer'.
+    3. Deploy a StatefulSet with 3 replicas, each requesting a 15Gi PVC using
+       the StorageClass from step 2.
+    4. Verify that all pods are scheduled onto the current node (since it’s the
+       only one with sufficient storage).
+    """
+
+    lht_hostId = get_self_host_id()
+    nodes = client.list_node()
+    for node in nodes:
+        if node.id != lht_hostId:
+            disks = node.disks
+            for _, disk in disks.items():
+                disk.storageReserved = disk.storageMaximum - 10 * Gi
+            update_disks = get_update_disks(disks)
+            update_node_disks(client, node.name, disks=update_disks,
+                              retry=True)
+
+    sc_name = 'longhorn-wait-for-first-consumer'
+    storage_class['metadata']['name'] = sc_name
+    storage_class['volumeBindingMode'] = 'WaitForFirstConsumer'
+    storage_class['parameters']['numberOfReplicas'] = '1'
+    create_storage_class(storage_class)
+
+    statefulset['spec']['replicas'] = 3
```

Comment:
Test case `test_storage_capacity_aware_pod_scheduling` failed on master-head (longhorn-manager c933a9e).

Only `1` replica of the statefulset can be successfully running, since each of them will create a `15Gi` pvc, but only the first one will have sufficient storage space.

Reply:
Hi @yangchiu. In your testing environment, what is the available storage size on the self-host node (the node where the test pod is running)? This test case currently assumes at least 45Gi of available storage, but I realize I shouldn’t make that assumption.

Reply:
@yangchiu, I've reduced the required storage. Could you please run the test again?